bash_automation.md

```diff
@@ -0,0 +1,165 @@
+# Automating workflows using bash
+
+## Objectives
+
++ Write a shell script with multiple variables.
++ Incorporate a for loop into a shell script.
++ Understand how to make a workflow more efficient and less error-prone
+
+
+We have now worked through two workflows, RNA-seq analysis and variant calling.
+In both cases, we pasted all of our commands into the terminal, and used for 
+loops to run each step of the workflow over our six samples. This is a great way
+to understand how each tool works and to establish a workflow you want to use.
+
+Let's imagine now that we decided to download all 672 samples from this dataset
+instead of only working with our six samples. We already know how each tool in
+our workflow works, and we know how to run them. We can now use bash scripting
+to automate our workflow. 
+
+We are going to automate the quality control portion of our workflow. To write 
+a script to run our FastQC analysis, we’ll take each of the commands we entered 
+to run FastQC and process the output files and put them into a single file with 
+a `.sh` extension. The `.sh` is not essential, but serves as a reminder to 
+ourselves and to the computer that this is a shell script.
+
+## Starting with automation
+
+We will be constructing our bash script in RStudio. We will construct our bash
+script in a text file in the "script" pane of RStudio, and we will execute it 
+in the "Terminal" tab of the "console" pane. We could use nano, but RStudio
+makes it easier to navigate a text file. 
+
+First, let's make a new directory called `scripts`.
+
+```
+cd ~
+mkdir scripts
+```
+
+## Constructing a bash script
+
+When we write a bash script, we need to add *all* commands that we ran in our
+workflow. This includes making and changing directories, moving files, and 
+running analysis programs like `fastqc`. 
+
+Let's start by changing into the right directory and running `fastqc`:
+
+```
+cd ~/data/
+
+fastqc *fastq.gz
+```
+
+That's it! We now have a bash script that automates `fastqc`! 
+
+Let's run the bash script from our terminal in RStudio
+
+```
+bash qc.sh
+```
+
+Let's add more to our script. Next we'll organize our output files. 
```

Comment:
Oh i think i see what's going on. This is supposed to be taking off from working in the RStudio environment, and being entered in the RScript window? That may be unnecessarily confusing

I think it might be better able to be a standalone lesson, and we'll just put the same intro at the top (about accessing your jetstream if you don't have it), and then running a download command of data if we don't have it, like the start of the diff expression and viz doc. Then we can have them work at the command line instead of R for this. 

But let me know if you have strong feelings about keeping it in R!

Reply:
Nope, I'm fine with this. Again though, I think it's worth it to keep it standard across bash automation and snakemake, so @sateeshperi please way in on your preferences!